source/views/menus/menu-bonapp.js

```diff
@@ -217,9 +217,11 @@ export class BonAppHostedMenu extends React.PureComponent<Props, State> {
 		const dayparts = cafeMenu.days[0].cafe.dayparts
 
 		// either use the meals as provided by bonapp, or make our own
-		const mealInfoItems = dayparts[0].length ? dayparts[0] : DEFAULT_MENU
+		const mealInfoItems =
+			dayparts.length !== 0 && dayparts[0].length ? dayparts[0] : DEFAULT_MENU
```

Comment:
What happens if `dayparts.length === 0`? Then this becomes `null` or `undefined`, I guess. Is that handled?

Reply:
If `dayparts.length === 0`, the ternary "short-circuits" and doesn't bother checking the second condition and the condition is evaluated as `false`. Right? @rye

Reply:
Well for one, I don't think this logic covers the case where `dayparts[0]` doesn't have a `.length`?

In any case, here's what I get when `dayparts.length === 0`:

```console
> let ary = []
undefined
> ary.length !== 0 && ary[0].length ? ary[0] : 'default'
'default'
```

***

Ohh… precedence here means the condition on the ternary is `dayparts.length !== 0 && dayparts[0].length`, not `dayparts[0].length`.

You're right, the `&&` stops evaluating after the first condition becomes `false`.

My new question is: can you wrap the condition in parens? (Will prettier change it back?) It's more readable that way.

Reply:
Prettier will remove the parens.

Reply:
If `dayparts[0]` doesn't have a `.length`, that part of the condition will return `undefined`, which will make the whole condition evaluate as `undefined` and so the ternary will return the default schedule, as desired.